server/src/main/java/org/candlepin/model/dto/ProductData.java

```diff
@@ -1001,7 +1072,8 @@ public int hashCode() {
             .append(this.content)
             .append(this.dependentProductIds)
             .append(this.branding)
-            .append(this.locked);
+            .append(this.locked)
+            .append(ppHashCode);
```

Comment:
Same here, no need for calculating hashcode and setting it. Just `append.(this.getProvidedProducts())`